test/supplemental_genbank_tests/genbank_upload_full_test.py

```diff
@@ -167,3 +171,125 @@ def test_simple_shock_upload(self):
         self.assertIsNotNone(result['genome_ref'])
         self.assertTrue(int(
             result['genome_info'][10]['Number of Protein Encoding Genes']) > 0)
+
+    def _run_test_fail(self, params, error_message, mass=True):
+        with pytest.raises(Exception) as got:
+            if mass:
+                self.getImpl().genbanks_to_genomes(self.getContext(), params)
+            else:
+                self.getImpl().genbank_to_genome(self.getContext(), params)
+        assert_exception_correct(got.value, ValueError(error_message))
+
+    def _get_objects_data(self, refs):
+        results = self.getDfuClient().get_objects({'object_refs': refs})
+        return [result["info"] for result in results["data"]]
+
+    def test_genbank_to_genome_invalid_workspace(self):
```

Comment:
There a large number of parameters to genbanks_to_genomes that aren't tested at all here

Reply:
Like what?  `generate_ids_if_needed` and `generate_missing_genes`

Reply:
many of these: https://github.com/kbaseapps/GenomeFileUtil/pull/208/files#diff-4cf0f8eeebf002332191c051d29b71d2f498f31b2cf3af852e3368b4bd92f204R75-R85

Reply:
I need more clarification here. 
Do we want to test their invalid cases and then assert valuerError or 
pass these optional params to genbanks_to_genomes, and then check rtn info is correct and as expected based on the uploaded data

Reply:
If a method is added to an API with a set of parameters, those parameters need tests that validate they behave correctly. In some cases, it might be as simple as checking that the parameter value winds up in the object data saved to the workspace for its default and set values. Other cases may be more complicated based on the behavior of the code dealing with the parameter, including error conditions, specific parameter values triggering different code, etc.

Reply:
This is a case where I think an argument could be make for a separate PR for the parameters fwiw

Reply:
That would be great. I will mark it as TODO then.

Reply:
👍

Reply:
:+1: